### PR TITLE
Update Pinned CI Packages

### PR DESCRIPTION
**Pull request by scbedd**
This introduces new `pylint` failures.

Need to get these resolved before merge.

EDIT 7/8: Updated with latest master (including `virtualenv` pins)

### CODE REVIEW

**Comment by chlowell** [comment]
I see a new exception in `pylint_guidelines_checker.py` when analyzing azure-keyvault-keys:
```
Traceback (most recent call last):
  ...
  File "azure-keyvault-keys\.tox\lint\lib\site-packages\pylint\__main__.py", line 18, in <module>
    pylint.run_pylint()
  File "azure-keyvault-keys\.tox\lint\lib\site-packages\pylint\__init__.py", line 22, in run_pylint
    PylintRun(sys.argv[1:])
  File "azure-keyvault-keys\.tox\lint\lib\site-packages\pylint\lint\run.py", line 344, in __init__
    linter.check(args)
  File "azure-keyvault-keys\.tox\lint\lib\site-packages\pylint\lint\pylinter.py", line 871, in check
    self.get_ast, self._iterate_file_descrs(files_or_modules)
  File "azure-keyvault-keys\.tox\lint\lib\site-packages\pylint\lint\pylinter.py", line 904, in _check_files
    self._check_file(get_ast, check_astroid_module, name, filepath, modname)
  File "azure-keyvault-keys\.tox\lint\lib\site-packages\pylint\lint\pylinter.py", line 930, in _check_file
    check_astroid_module(ast_node)
  File "azure-keyvault-keys\.tox\lint\lib\site-packages\pylint\lint\pylinter.py", line 1063, in check_astroid_module
    ast_node, walker, rawcheckers, tokencheckers
  File "azure-keyvault-keys\.tox\lint\lib\site-packages\pylint\lint\pylinter.py", line 1107, in _check_astroid_module
    walker.walk(ast_node)
  File "azure-keyvault-keys\.tox\lint\lib\site-packages\pylint\utils\ast_walker.py", line 75, in walk
    self.walk(child)
  File "azure-keyvault-keys\.tox\lint\lib\site-packages\pylint\utils\ast_walker.py", line 72, in walk
    callback(astroid)
  File "azure-sdk-for-python\scripts\pylint_custom_plugin\pylint_guidelines_checker.py", line 547, in visit_classdef
    msg_id="client-incorrect-naming-convention", node=node, confidence=None
TypeError: add_message() got an unexpected keyword argument 'msg_id'
```

**Comment by kristapratico** [comment]
> I see a new exception in pylint_guidelines_checker.py when analyzing azure-keyvault-keys:

This sounds like my fault. I'll look into what changed with the custom checkers for the newer version of pylint.

**Comment by kristapratico** [comment]
@chlowell pushed a commit fixing that error, you _should_ be unblocked now.

**Comment by chlowell** [comment]
Thanks @kristapratico, that appears to have worked.

I pushed a couple commits, one to address warnings in azure-keyvaut-keys and azure-identity, and one to disable the `import-outside-toplevel` rule.

File: eng/test_tools.txt [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
 pyyaml==5.3.1
 packaging==20.4
 wheel==0.34.2
-Jinja2==2.11.1
+Jinja2==2.11.2
 
 # Locking pylint and required packages
-pylint==2.3.1; python_version >= '3.4'
 pylint==1.8.4; python_version < '3.4'
-astroid==2.3.3; python_version >= '3.4'
```
Comment (praveenkuttappan) [inline eng/test_tools.txt:26]: This is intentional to install latest version of astroid that is required by pylint. right?

File: sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_internal/algorithms/aes_cbc.py [hunk 2]
```diff
@@ -39,7 +39,6 @@ def finalize(self, data):
         padded = self._ctx.finalize()
         return self._padder.update(padded) + self._padder.finalize()
 
-    @property
```
Comment (praveenkuttappan) [inline sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_internal/algorithms/aes_cbc.py:42]: Assuming that it is intentional change to remove property flag? these look like private classes only anyway

Comment (scbedd) [reply sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_internal/algorithms/aes_cbc.py:42]: yes

File: sdk/storage/azure-storage-blob/azure/storage/blob/aio/_blob_service_client_async.py [hunk 3]
```diff
@@ -3,7 +3,7 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
-
+# pylint: disable=invalid-overridden-method
```
Comment (rakshith91) [inline sdk/storage/azure-storage-blob/azure/storage/blob/aio/_blob_service_client_async.py:6]: This does not solve the purpose. We must create a common base class for sync and async and inherit both the clients from the base class.


```
class BlobServiceClientBase():

class BlobServiceClient(BlobServiceClientBase):

again in ./aio
class BlobServiceClient(BlobServiceClientBase):
```

Comment (xiafu-msft) [reply sdk/storage/azure-storage-blob/azure/storage/blob/aio/_blob_service_client_async.py:6]: The mypy pr is already doing this, I think we can reuse that one
Also this problem happened for upload, download, policies, I don't think we need to create a base and inherit  for sync and async for all cases.

**Pull request merged**